src/layers/microscopy-viewer-layer/data-utils/zarr-utils.js

```diff
@@ -1,56 +1,51 @@
-import { slice, openArray } from 'zarr';
-
-async function getData({ arr, channel, x, y, stride, tilingWidth }) {
-  const arrSlice = slice(
-    stride * tilingWidth * y + stride * x,
-    stride * tilingWidth * y + stride * (x + 1)
-  );
-  const dataSlice = await arr.get([arrSlice]);
-  const { data } = dataSlice;
-  const dataObj = {};
-  dataObj[channel] = data;
-  return dataObj;
+/* eslint-disable import/no-extraneous-dependencies */
+import { openArray } from 'zarr';
+
+function decodeChannels({ data, shape }) {
+  const offset = data.length / shape[0];
+  const tileData = [];
+  for (let i = 0; i < shape[0]; i += 1) {
+    const channelData = data.subarray(offset * i, offset * i + offset);
+    tileData.push(channelData);
+  }
```

Comment:
Maybe you can explain in person again how we know the order here again as I can't remember.

Reply:
Ordering comes from how the layers were encoded along the 3dim of the TIFF. The TIFF I was provided had no metadata so I don't know what the labels are. I know the DAPI label is correct, but I don't know about the others. Fortunately we can add better mappings to correct channels within this `utils` in the future.